File: number_of_arithmetic_slices/numberOfArithmeticSlices.js
const numberOfArithmeticSlices = (nums) => {
	let answer = 0;
	for (let i=0; i<nums.length-2;i++) {
		if (nums[i]-nums[i+1] == nums[i+1]-nums[i+2]) {
			answer += 1
			for (let j=(i+3); j < nums.length; j++) {
				if (nums[j-1]-nums[j] == nums[i]-nums[i+1]) {
					answer += 1;
				} else {
					break;
				}
			}
		}
	}
	return answer;
};

File: number_of_arithmetic_slices/number_of_arithmetic_slices.rb
def number_of_arithmetic_slices(nums)
  answer = 0
  (0...nums.length-2).each do |idx|
    if nums[idx] - nums[idx+1] == nums[idx+1]-nums[idx+2]
      answer += 1
      (idx+3...nums.length).each do |idx2|
        if nums[idx2-1]-nums[idx2] == nums[idx] - nums[idx+1]
          answer += 1
          else
          break
        end
      end
    end
  end
  answer
end
